gloo/test/allreduce_test.cc

```diff
@@ -26,8 +26,13 @@ namespace {
 
 // RAII handle for aligned buffer
 template <typename T>
+#ifdef _WIN32
```

Comment:
What issue was the aligned allocator causing?

Reply:
Probably fine for now, but in the future `aligned_allocator` can be implemented using `_aligned_malloc` and `_aligned_free` on Windows, since it doesn't have `posix_memalign`.